whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotObjectSynchronizer.java

```diff
@@ -68,14 +73,19 @@ NrtmVersionInfo initializeSnapshotObjects(final NrtmSource source) {
                             continue;
                         }
                         final RpslObject dummyRpsl = dummifierNrtm.dummify(NRTM_VERSION, rpslObject);
-                        batch.add(new SnapshotObject(0, version.getId(), object.objectId(), object.sequenceId(), dummyRpsl));
+                        final NrtmSourceModel source = sourceMap.get(dummyRpsl.getValueForAttribute(AttributeType.SOURCE));
+                        if (source == null) {
+                            final String msg = "RPSL object declares an unknown source attribute";
+                            LOGGER.error(msg + " " + dummyRpsl.getValueForAttribute(AttributeType.SOURCE));
```

Comment:
This whole logic should be in nrtmSource component, you should just pass the source name and it gives you id.

Reply:
agree. i'll try it out.

Reply:
still in place for now, but the next story doesn't use the db for snapshot objects so this class will be removed.